lib/shopify_dashboard_plus.rb

```diff
@@ -102,35 +130,43 @@ def validate_date_range(from, to)
 
 
   ## Metrics Helpers
-  def get_total_revenue(orders)
-    revenue = orders.collect{|order| order.total_price.to_f }.inject(:+).round(2) rescue 0
-    revenue ||= 0
+  def display_as_currency(value)
+    ShopifyAPI::Shop.current.money_with_currency_format.gsub("{{amount}}", value.to_s) rescue "N/A"
+  end
+
+  def get_date_range(first, last)
+    DateTime.parse(last).mjd - DateTime.parse(first).mjd
+  end
+
+  def get_total_revenue(collections)
+    # collections is an array of ActiveRecord::Collections, each containing orders
+    collections.collect{|orders| orders.collect{|order| order.total_price.to_f }.inject(:+).round(2) }.inject(:+).round(2) rescue 0
   end
```

Comment:
A few points

- `map` is more common than `collect`, if only to save a few characters
- [`Enumerable#flat_map`](http://ruby-doc.org/core-2.2.2/Enumerable.html#method-i-flat_map) will let you build an array of all the totals and add them once
- I'd separate your error handling to make it clear you're falling back to `0` if anything bad happens
- You only need to round once rather than at every step (I think)

which gives you something like

```ruby
def get_total_revenue(collections)
  totals = collections.flat_map do |orders| 
    orders.map { |order| order.total_price.to_f }
  end
  totals.inject(:+).round(2)
rescue
  0
end
```

That said seen as you're just calculating a total, building a bunch of intermediate arrays isn't the most efficient approach. Using [`Enumerable#reduce`](http://ruby-doc.org/core-2.2.2/Enumerable.html#method-i-reduce) would be faster albeit harder to read:

```ruby
def get_total_revenue(collections)
  collections.reduce(0) do |total, orders|
    total + collection_total(orders)
  end.round(2)
rescue
  0
end

def collection_total(orders)
  orders.reduce(0) do |total, order| 
    total + order.total_price.to_f
  end
end
```